roles/hxr.admin-tools/defaults/main.yml

```
@@ -15,7 +15,7 @@ admin_packages:
   - net-tools
   - unzip
   - mutt
-  - byobu
+  - "{{ byobu if ansible_distribution_major_version < '9' | default(omit) }}"
```

```suggestion
  - "{{ 'byobu' if ansible_distribution_major_version < '9' else omit }}"
```

Just run this task to see why.

```
    - name: Debug.
      ansible.builtin.debug:
        msg: "{{ byobu if ansible_distribution_major_version < '9' | default(omit) }}"
```

---

The task runs for me, but yes we can change that

---

> The task runs for me, but yes we can change that

It runs because you are running the task in EL 9, so the variable `byobu` is never evaluated. If you run it on sn06 (running EL 8) you get this.

```
TASK [Debug.] ************************************************************************************************************************************************************************
fatal: [sn06.galaxyproject.eu]: FAILED! => {"msg": "The task includes an option with an undefined variable. The error was: 'byobu' is undefined\n\nThe error appears to be in '/home/jose/Dokumente/Repositories/usegalaxy-eu_infrastructure-playbook/htcondor.yml': line 5, column 7, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n  pre_tasks:\n    - name: Debug.\n      ^ here\n"}
```

Apart from that, `"{{ 'byobu' if ansible_distribution_major_version < '9' }}"` evaluates to an empty string when the condition after the if clause is not met, **not to an undefined variable**. Therefore the "default" filter does not come into action. You can realize this because the task

```yaml
    - name: Debug.
      ansible.builtin.debug:
        msg: "{{ 'byobu' if ansible_distribution_major_version < '8' | default(omit) }}"
```

yields 

```
TASK [Debug.] ************************************************************************************************************************************************************************
ok: [sn06.galaxyproject.eu] => {
    "msg": ""
}
```

on the headnode, but 

```yaml
    - name: Debug.
      ansible.builtin.debug:
        msg: "{{ 'byobu' if ansible_distribution_major_version < '8' else omit }}"
```

yields

```
TASK [Debug.] ************************************************************************************************************************************************************************
ok: [sn06.galaxyproject.eu] => {
    "msg": "Hello world!"
}
```

---

ok